### PR TITLE
Selection for equivariant tensors, and return Hausdorff distances

### PR DESCRIPTION
**Pull request by jwa7**
- Modifies the `GreedySelector.fit` method to allow for TensorMaps with a single components axis (i.e. equivariants)
- Allows the user to pass either an `int` or `dict` for `n_to_select`. Setting to an int sets a fixed value for every block. Setting to -1 sets to the maximum number of samples/features for each block. Setting to a dict allows explicit control of the number of selection points for each block
- Returns the sorted Haussdorf distances in the `selector.get_select_distance` property.

For instance, one can plot sorted Haussdorf distances for the features of each block as follows:

```python
import matplotlib.pyplot as plt
import numpy as np

import equistore
from equisolve.numpy import feature_selection

tensor = equistore.load("tensor.npz")

# n_to_select = 10  # set num features to 10 for every block
n_to_select = -1  # set to the max for each block
# n_to_select = {  
#     tuple(key.values): len(block.properties) - np.random.randint(0, 2)
#     for key, block in tensor.items()
# }  # control for each block

selector = feature_selection.FPS(n_to_select=n_to_select, initialize="random")
tensor_fps = selector.fit_transform(tensor)

fig, ax = plt.subplots()
for key, block in selector.get_select_distance.items():
    ax.loglog(np.sort(block.values.flatten())[::-1], label=key)

fig.legend()
```

Still to do:

- [x] Update tests

----
:books: Documentation preview :books:: https://equisolve--59.org.readthedocs.build/en/59/

### CODE REVIEW

**Comment by jwa7** [comment]
> > (NB: Haussdorf is spelled as such in the code (and not Hausdorff) for consistency with the corresponding skmatter methods)
> 
> We will fix this [scikit-learn-contrib/scikit-matter#207](https://github.com/scikit-learn-contrib/scikit-matter/pull/207)

Ok thanks, I will correct the spelling in this PR too.

**Comment by PicoCentauri** [comment]
What is the status here @jwa7 ? Seems like there some attribute problems?

File: src/equisolve/numpy/_selection.py [hunk 1]
```diff
@@ -57,45 +60,153 @@ def support(self) -> TensorMap:
...
             Training vectors.
         :param warm_start:
-            Whether the fit should continue after having already run, after increasing
-            `n_to_select`. Assumes it is called with the same X.
+            Whether the fit should continue after having already run, after
+            increasing `n_to_select`. Assumes it is called with the same X.
         """
-        if len(X.components_names) != 0:
-            raise ValueError("Only blocks with no components are supported.")
+        # CHeck that we have only 0 or 1 comoponent axes
```
Comment (PicoCentauri) [inline src/equisolve/numpy/_selection.py:87]:
```suggestion
        # Check that we have only 0 or 1 comoponent axes
```

**Review by PicoCentauri — CHANGES_REQUESTED** [review]
I got a quick look. Just found a minor typo.

**Review by PicoCentauri — APPROVED** [review]
LGTM. Merge when you want and the CI passed.

**Comment by jwa7** [comment]
LGTM => looks good to me :handshake: let’s get this merged
